src/tools/compiletest/src/runtest.rs

```diff
@@ -1488,66 +1488,23 @@ impl<'test> TestCx<'test> {
-            let aux_output = TargetLocation::ThisDirectory(self.aux_output_dir_name());
-            let aux_cx = TestCx {
-                config: self.config,
-                props: &aux_props,
-                testpaths: &aux_testpaths,
-                revision: self.revision,
-            };
-            // Create the directory for the stdout/stderr files.
-            create_dir_all(aux_cx.output_base_dir()).unwrap();
-            let mut aux_rustc = aux_cx.make_compile_args(&aux_testpaths.file, aux_output);
-
-            let crate_type = if aux_props.no_prefer_dynamic {
-                None
-            } else if self.config.target.contains("cloudabi")
-                || self.config.target.contains("emscripten")
-                || (self.config.target.contains("musl") && !aux_props.force_host)
-                || self.config.target.contains("wasm32")
-            {
-                // We primarily compile all auxiliary libraries as dynamic libraries
-                // to avoid code size bloat and large binaries as much as possible
-                // for the test suite (otherwise including libstd statically in all
-                // executables takes up quite a bit of space).
-                //
-                // For targets like MUSL or Emscripten, however, there is no support for
-                // dynamic libraries so we just go back to building a normal library. Note,
-                // however, that for MUSL if the library is built with `force_host` then
-                // it's ok to be a dylib as the host should always support dylibs.
-                Some("lib")
-            } else {
-                Some("dylib")
-            };
+            build_auxiliary(self, rel_ab, &aux_dir);
+        }
 
-            if let Some(crate_type) = crate_type {
-                aux_rustc.args(&["--crate-type", crate_type]);
-            }
+        for rel_ab in &self.props.aux_crates {
+            build_auxiliary(self, &rel_ab.value, &aux_dir);
+        }
 
-            aux_rustc.arg("-L").arg(&aux_dir);
+        rustc.envs(self.props.rustc_env.clone());
 
-            let auxres = aux_cx.compose_and_run(
-                aux_rustc,
-                aux_cx.config.compile_lib_path.to_str().unwrap(),
-                Some(aux_dir.to_str().unwrap()),
-                None,
-            );
-            if !auxres.status.success() {
-                self.fatal_proc_rec(
-                    &format!(
-                        "auxiliary build of {:?} failed to compile: ",
-                        aux_testpaths.file.display()
-                    ),
-                    &auxres,
-                );
-            }
+        for aux_crate in &self.props.aux_crates {
+            rustc.arg("--extern");
+            rustc.arg(format!("{}={}/lib{}",
+                              aux_crate.key,
+                              aux_dir.display(),
+                              &aux_crate.value.replace(".rs", ".so").replace("-","_")));
```

Comment:
@alexcrichton is there a place on the compiler where I can get the **official** codepath that does this in a cross-platform way?

Reply:
I think this can probably use `std::env::consts::DLL_EXTENSION` perhaps? We typically don't need to factor in cross compilation here